### PR TITLE
[custom_op] setup_context fills in default values

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #124852
* #124806
* #124805
* #124637

This is to mirror autograd.Function's setup_context behavior.
The PyTorch Dispatcher removes default values for "FC/BC reasons", but I
convinced myself there's no FC/BC problem for the setup_context API.

Test Plan:
- new tests

### CODE REVIEW

File: torch/_library/utils.py [hunk 1]
```diff
@@ -158,6 +158,24 @@ def mutates_and_returns_first_arg(op: torch._ops.OpOverload):
     return True
 
 
+def fill_defaults(schema, args, kwargs):
```
Comment (albanD) [inline torch/_library/utils.py:161]: We don't already have a version of that somewhere? :o 
I'm sure this is something that should be in our subclass utils :p

**Review by albanD — APPROVED** [review]
Yep!

File: torch/_library/utils.py (hunk 1 above)
Comment (zou3519) [reply torch/_library/utils.py:161]: I have not been able to find one. There's something similar for torch_function and ONNX but those are not exactly what we want

**Pull request closed without merging**